src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypeHandler.cs

```diff
@@ -74,6 +74,11 @@ public virtual Task Rename(string connectionUri, string objectUrn, string newNam
                     throw new Exception(SR.ObjectNotRenamable(objectUrn));
                 }
             }
+            // If connection is not instantiated in the using statement, it can be leaked here, therefore close explicitly.
```

Comment:
I'm not sure I understand this - using should always call Dispose on the object. Is there something special about these connections that make it so they don't actually get closed even when Disposed?

Sample I was playing around with here : https://dotnetfiddle.net/A8LWSE

Reply:
Here's a quick repro, and on debug I found this was happening too..

![image](https://github.com/microsoft/sqltoolsservice/assets/13396919/26ea0a4b-8e22-49d9-879a-d0ee3e22341a)

Reply:
So you're saying somewhere in the using statement the actual underlying connection object is being swapped out? With the result that the old one is disposed (and closed) correctly, but the new one isn't?